syntax = "proto3";

package post;

message Post {
    string id = 1;
    string title = 2;
    string description = 3;
    string creator_id = 4;
    string created_at = 5;
    string updated_at = 6;
    bool is_private = 7;
    repeated string tags = 8;
}

message CreatePostRequest {
    string title = 1;
    string description = 2;
    string creator_id = 3;
    bool is_private = 4;
    repeated string tags = 5;
}

message UpdatePostRequest {
    string id = 1;
    string title = 2;
    string description = 3;
    string creator_id = 4;
    bool is_private = 5;
    repeated string tags = 6;
}

message DeletePostRequest {
    string id = 1;
    string creator_id = 2;
}


message DeleteResponse {
    bool success = 1;
}
  
message GetPostRequest {
    string id = 1;
    string user_id = 2;
}
  
message ListPostsRequest {
    int32 page = 1;
    int32 page_size = 2;
    string user_id = 3;
}

message ListPostsResponse {
    repeated Post posts = 1;
    int32 total = 2;
    int32 page = 3;
    int32 page_size = 4;
}

service PostService {
    rpc CreatePost(CreatePostRequest) returns (Post) {}
    rpc UpdatePost(UpdatePostRequest) returns (Post) {}
    rpc DeletePost(DeletePostRequest) returns (DeleteResponse) {}
    rpc GetPost(GetPostRequest) returns (Post) {}
    rpc ListPosts(ListPostsRequest) returns (ListPostsResponse) {}
}
